### DISCUSSION TITLE
Needing help on duplicate data

### DISCUSSION THREAD

**Discussion by seancrossettie** [nss-evening-cohort-14/evening-client-side#300] [Capstones!] [answered]
Sean Rossettie

## issue
The issue I am having is that I am able to save duplicates of the same artists and events to each user. I don't want to be able to follow multiple of the same artists and events as that would be clunky. I have a general idea as how to handle multiples as far as ordering by eventId or artistId, which are unique to each event and artist and then perhaps pushing those into a new Set to eliminate duplicates, but I am a little hazy as far as how to feasibly do this since the Set will be looking at the names of the objects, which are firebaseKeys and totally unique to each entry in firebase as opposed to eventId/artistId, which are unique in the Songkick API but not firebase. 

Im wondering if this is a Promise.all situation, but am not sure where to start.

## error messages
no error messages appear in this case.

## What have you googled?
https://firebase.google.com/docs/database/security
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

## Link to file in repo
https://github.com/seancrossettie/show-finder/blob/remove-duplicates/src/components/App/App.js

**Comment by drteresavasquez** [comment]
@seancrossettie yes, a promise.all would help and then filtering through to only show the ones that you want to show is the best way to handle this.

**Reply by seancrossettie** [reply]
Okay, Im working on this right now and have been able to write a promise to get all of my artist id's, but am not sure how to implement the Promise.all() to create a new set of objects that have the kvp of followed. Im lost on how to go about comparing the two sets of data.

**Reply by drteresavasquez** [reply]
You need to use some array methods and create a new array of data. Try setting everything up and work from there. Tell us how far you get.

**Reply by seancrossettie** [reply]
#### API calls
https://github.com/seancrossettie/show-finder/blob/compare-artists/src/helpers/data/artistData.js
#### Search form
https://github.com/seancrossettie/show-finder/blob/compare-artists/src/components/search%20page/SearchForm.js

Ive gotten to the point to where I am able to create both an array of my saved artist's ids and an array of the searched artist's ids. Im assuming that after that I need to compare the two but I am not sure how. Also, Im a bit lost on what to do with the comparison as far as changing the ui to reflect whether or not I have already followed an artist that matches one of my artists

**Comment by drteresavasquez** [comment, marked as answer]
Check this out: https://github.com/nss-evening-cohort-14/evening-client-side/discussions/311

**Reply by seancrossettie** [reply]
This ticket solved my issue

**Reply by seancrossettie** [reply]
```
const compareArtists = (user, artist) => new Promise((resolve, reject) => {
  Promise.all([getArtists(artist), getMyArtists(user)])
    .then(([searchArtistArr, myFollowed]) => {
      resolve(searchArtistArr.map((artistObj) => ({
        ...artistObj,
        isFollowed: Boolean(myFollowed.find((followed) => followed.artistId === artistObj.id))
      })));
    })
    .catch((error) => reject(error));
```
